flepimop/gempyor_pkg/src/gempyor/NPI/SinglePeriodModifier.py

```diff
@@ -145,6 +147,7 @@ def __createFromDf(self, loaded_df, npi_config):
         self.affected_subpops = set(self.subpops)
         if npi_config["subpop"].exists() and npi_config["subpop"].get() != "all":
             self.affected_subpops = {str(n.get()) for n in npi_config["subpop"]}
+            self.affected_subpops = list(set(self.affected_subpops).intersection(self.subpops))
```

Comment:
again, code to potentially DRY out

Reply:
yes, absolutely. Trying to keep changes to signature minimal for now